### PR TITLE
HEEDLS-891 Add PRN to delegate upload journey

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-891

### Description
Added PRN fields to the delegate upload journey.

### Screenshots
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by JBsoftwire — CHANGES_REQUESTED** [review]
Found a typo and a minor tweak in the tests.

File: DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs [hunk 1]
```diff
@@ -129,6 +143,18 @@ public bool Validate(IEnumerable<int> allowedJobGroupIds)
...
+                Error = BulkUploadResult.ErrorReason.HasPrnButMissingPrnValue;
+            }
+            else if (!string.IsNullOrEmpty(Prn) && (Prn.Length < 5 || Prn.Length > 20))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidPrnLength;
+            }
+            else if (!string.IsNullOrEmpty(Prn) && !PrnRegex.IsMatch(Prn))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidPrnCharacters;
+            }
```
Comment (livzorn) [inline DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs:157]: So we don't care if HasPRN is False and a PRN is provided?

**Review by livzorn — COMMENTED** [review]
Just a few small questions/comments

File: DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs (hunk 1 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs:157]: Unless HasPRN is true we're saving null to the database, so I don't think it needs a validation case.

**Pull request merged**